### PR TITLE
Use sed over nodejs

### PR DESCRIPTION
**Pull request by lajp**
node sux, dont use it, it's bloat, use sed (one line)

### CODE REVIEW

**Comment by ahnl** [comment]
I don't like it.

**Comment by niilasoika** [comment]
I see how this could work in sort of "integrated" enviromnents, but for an average developer I think that `npx env-exampler` is a lot easier to remember, than going to github to copy-paste a command. That's my opinion.

**Comment by DrVilepis** [comment]
I like this pull pr, this has no external dependencies outside of coreutils, is much more readable and portable, as well as being marginally faster likely. The command is also much easier to remember, as you can just run exampler anywhere by adding it to your path.

File: README.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ env-exampler is a script that generates a .env.example file based on the current
 
 ## Usage
 
-Run `npx env-exampler` in the directory with your .env file.
+Run `./exampler` in the directory with your .env file.
```
Comment (ahnl) [inline README.md:7]: You'll have to keep a local copy of the script. The npx solution is imo nicer:
- cross platform, also doesn't break the git hook on Windows machines
- all you need is npm, one short command to get it&run it
- though it's probably slower

**Comment by ahnl** [comment]
> I don't like it.

Actually I like it, it's short and sweet, but see my concerns above.

**Comment by Esinko** [comment]
Speed of execution is irrelevant here. Portability and ease of use should be the focus.
Here's an idea though: Why not both?
Just leave a note like: "psst... Linux enthusiasts may add this script to their path" etc.

File: README.md (hunk 1 above)
Comment (DrVilepis) [reply README.md:7]: I disagree with this, sed is standard for all posix compliant systems, also does not require an internet connection, and as above stated, does not require even npm on your system.

**Comment by lajp** [comment]
I understand your concerns... Perhaps you could have the hook do something along the lines of `curl -Ls https://raw.githubusercontent.com/lajp/env-exampler/master/exampler | sh` 

Although that only solves the convenience for non-windows users.

File: README.md (hunk 1 above)
Comment (Esinko) [reply README.md:7]: > cross platform, also doesn't break the git hook on Windows machines

sed does not ship with Windows, which is still arguably the most used OS among common developers.

**Comment by lajp** [comment]
I like the idea of adding it to path, or then just adding an alias for it (since it's so simple)

**Comment by DrVilepis** [comment]
I dont think this is different for windows users, they can install some coreutils just like they can install npm. This takes no extra steps compared to the npm solution.

**Comment by kymppi** [comment]
But i bet that they already have npm installed

**Comment by Esinko** [comment]
> I dont think this is different for windows users, they can install some coreutils just like they can install npm. This takes no extra steps compared to the npm solution.

Most common developers already have NPM installed, so installing coreutils would be an extra step.
(+ documentation for installing and utilizing NPM is much more widespread)

**Comment by lajp** [comment]
common developer != nodejs developer.

**Comment by ahnl** [comment]
I bet most who will need env-exampler are Node.js developers.

**Comment by kymppi** [comment]
NodeJs is pretty common, so i guess common developer would have it.

**Comment by DrVilepis** [comment]
I actually still disagree, I would find to expect people using npm and people needing env vars a quite small overlap, as I have understood that node has things as config.json.

**Comment by Esinko** [comment]
> common developer != nodejs developer.

Most common programming languages:
1. Javascript
2. HTML/CSS
...

https://insights.stackoverflow.com/survey/2020#technology-programming-scripting-and-markup-languages-professional-developers

**Comment by ahnl** [comment]
I think it would be good to keep both. They are good in their own ways.

Maybe we should add the sed-script to README or point to a different repo?

**Comment by Esinko** [comment]
> Speed of execution is irrelevant here. Portability and ease of use should be the focus. Here's an idea though: Why not both? Just leave a note like: "psst... Linux enthusiasts may add this script to their path" etc.

I think this is the best way. Let's just do both.
Its own section in this repo's README is probably enough. Hiding it away just makes this whole conversation useless.

**Comment by DrVilepis** [comment]
> > common developer != nodejs developer.
> 
> Most common programming languages:
> 
>     1. Javascript
> 
>     2. HTML/CSS
>        ...
> 
> 
> https://insights.stackoverflow.com/survey/2020#technology-programming-scripting-and-markup-languages-professional-developers

This does not equal npm users, I personally use raw javascript and html/css with rust frameworks. Like [actix-web](https://actix.rs/)

**Comment by Esinko** [comment]
> This does not equal npm users, I personally use raw javascript and html/css with rust frameworks.

Arguably, it does. Most websites use some Javascript framework, which are often installed via NPM and Node.Js as a programming language is dependent on the existence of NPM.
\+ You are probably the least common JS user :p

Regardless of what is truthfully the most common JS user, most of them still just want simple steps for installation and won't bother with any extra fancy stuff.

**Comment by raikasdev** [comment]
.env file is most common with NodeJS projects with dotenv.

**Pull request closed without merging**